File: backend/src/Domain/Entities/TeacherAssignment.cs
namespace Domain.Entities;

public class TeacherAssignment
{
    public int Id {get; set;}
    
    public int TeacherId {get; set;}
    public User Teacher {get; set;} = null!;

    public int CourseId {get; set;}
    public Course Course {get; set;} = null!;

    public int SubjectId {get; set;}
    public Subject Subject {get; set;} = null!;

    public DateTime AssignedAt {get; set;} = DateTime.UtcNow;

    public ICollection<Assignment> Assignments {get; set;} = new List<Assignment>();
}

File: frontend/src/types/teacher-assignment.ts
export interface TeacherAssignment {
  id: number;

  teacherId: number;
  teacherName: string;

  courseId: number;
  courseCode: string;
  courseName: string;

  subjectId: number;
  subjectCode: string;
  subjectName: string;

  assignedAt: string;
}

export interface CreateTeacherAssignmentRequest {
  teacherId: number;
  courseId: number;
  subjectId: number;
}